### PR TITLE
make this module v14 compatible

### PR DESCRIPTION
**Pull request by jurgenweber**
## what
- Upgrade to support Terraform 0.14 and bring up to current Cloud Posse standard

## why
- Support Terraform 0.14

supersedes and closes #65

### CODE REVIEW

**Comment by esteban1983cl** [comment]
Hi, somebody can approve this PR please, thanks

File: context.tf [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #
 
 module "this" {
-  source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"
+  source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.22.0"
```
Comment (aknysh) [inline context.tf:22]:
```suggestion
  source  = "cloudposse/label/null"
  version = "0.22.0"
```

**Review by aknysh — DISMISSED** [review]
thanks @jurgenweber 
please see comments

File: context.tf (hunk 1 above)
Comment (jurgenweber) [reply context.tf:22]: ah moving to tf registry, cool

**Comment by esteban1983cl** [comment]
Hi team, in current release 0.23.0 we have the following:

```hcl
module "dns" {
  source = "git::https://github.com/cloudposse/terraform-aws-route53-cluster-hostname.git?ref=tags/0.7.0"

  enabled  = module.this.enabled && length(var.zone_id) > 0 ? true : false
  dns_name = var.dns_name == "" ? module.this.id : var.dns_name
  ttl      = 60
  zone_id  = var.zone_id
  records  = [local.dns_name]

  context = module.this.context
}
```

terraform-aws-route-cluster-hostname is supported for terraform 0.14.5 but not its context module:

https://github.com/cloudposse/terraform-aws-route53-cluster-hostname/blob/abd24152b4b9acd4a1f595db3591449139e03b6e/context.tf#L22

```hcl
module "this" {
  source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"

  enabled             = var.enabled
  namespace           = var.namespace
  environment         = var.environment
  stage               = var.stage
  name                = var.name
  delimiter           = var.delimiter
  attributes          = var.attributes
  tags                = var.tags
  additional_tag_map  = var.additional_tag_map
  label_order         = var.label_order
  regex_replace_chars = var.regex_replace_chars
  id_length_limit     = var.id_length_limit

  context = var.context
}
```

The module terraform-null-label release 0.19.2 has restrictions for terraform versions:

https://github.com/cloudposse/terraform-null-label/blob/b3a0a44e470884bc974b837ad72cdb4fdeccf2de/versions.tf#L1

```hcl
terraform {
  required_version = ">= 0.12.0, < 0.14.0"
}
```

Due this explanation I can't use terraform 0.14.x

```txt
Error: Unsupported Terraform Core version

  on .terraform/modules/efs_promo_orchestrator.dns.this/versions.tf line 2, in terraform:
   2:   required_version = ">= 0.12.0, < 0.14.0"

Module module.efs_promo_orchestrator.module.dns.module.this (from
git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2)
does not support Terraform version 0.14.3. To proceed, either choose another
supported Terraform version or update this version constraint. Version
constraints are normally set for good reason, so updating the constraint may
lead to other errors or unexpected behavior.
```

Somebody can upgrade dns module to get 0.8.0 release version for dns.
Thanks in advance.

**Comment by htplbc** [comment]
LGTM

**Review by htplbc — APPROVED** [review]

**Comment by esteban1983cl** [comment]
Thanks you for this support, have a beautiful christmas

**Review by maximmi — APPROVED** [review]

**Pull request merged**